kotlinlib/src/mill/kotlinlib/KotlinModule.scala

```diff
@@ -314,6 +315,13 @@ trait KotlinModule extends JavaModule { outer =>
 
   private[kotlinlib] def internalReportOldProblems: Task[Boolean] = zincReportCachedProblems
 
+  @internal
+  override def bspBuildTarget: BspBuildTarget = super.bspBuildTarget.copy(
```

Comment:
What's the relation between this `bspBuildTarget` and the `jvmBuildTarget` you introduced in https://github.com/com-lihaoyi/mill/pull/3681?

Reply:
`bspBuildTarget` is defined in `BspModule`, and was there before my PRs. It contains details about the module, passed to the BSP clients via the `workspace/buildTargets` request. It goes along `bspBuildTargetData`, that contains language-specific data passed along in the response to this request too.

`bspJvmBuildTarget` contains JVM-specific details about the module. These are passed to clients via `bspBuildTargetData` for `JavaModule`, and as a field of what's passed in `bspBuildTargetData` for `ScalaModule` (same for `KotlinModule` in https://github.com/com-lihaoyi/mill/pull/3644). So `ScalaModule` (and `KotlinModule` in https://github.com/com-lihaoyi/mill/pull/3644) also use `bspJvmBuildTarget`, but uses it differently in `bspBuildTargetData`.